api/modules/users/users.service.js

```diff
@@ -0,0 +1,106 @@
+const { responseHelper } = require("../../helper/responses.helper");
+const { gameResultsService } = require("../game-results/game-results.service");
+const {
+  gameSettingsService,
+} = require("../game-settings/game-settings.service");
+const { leaderBoardsService } = require("../leaderboards/leaderboards.service");
+const { buildResponse } = require("../response/responseUtils");
+
+module.exports.usersService = {
+  createUser: (db, username) => {
+    let sql = "INSERT INTO users(user_name) values(?)";
+    return new Promise((resolve, reject) => {
+      db.run(sql, [username], (err, res) => {
+        if (err) {
+          reject(err);
+        } else {
+          resolve(res);
+        }
+      });
+    })
+      .then(() => {
+        return this.usersService.getUser(db, username).then((data) => {
+          const userId = data.userId;
+          return Promise.all([
+            leaderBoardsService.createUserHighScoreEntry(db, userId),
```

Comment:
There seems to be two things happening here, but the second could fail. Have you heard of transactions?